pennylane/compiler/qjit_api.py

```diff
@@ -17,43 +17,53 @@
+        compiler is supported. The Catalyst compiler works with the JAX interface
 
+        For more details, please see the Catalyst documentation and :func:`catalyst:.qjit`
+        docstring.
 
-    This decorator enables both just-in-time and ahead-of-time compilation,
-    depending on the compiler package and whether function argument type hints
-    are provided.
+    .. note::
+
+        Catalyst supports compiling QNodes that use ``lightning.qubit``,
+        ``lightning.kokkos``, ``braket.local.qubit``, and ``braket.aws.qubit``
+        devices. It does not support ``default.qubit``.
+
+        Please see the Catalyst documentation for more details on supported
+        devices, operations, and measurements.
 
     Args:
-        compiler(str): name of the compiler package (default value is ``catalyst``)
-        fn (Callable): the quantum or classical function
+        compiler (str): name of the compiler to use for just-in-time compilation
+        fn (Callable): the quantum or classical function to compile
         autograph (bool): Experimental support for automatically converting Python control
             flow statements to Catalyst-compatible control flow. Currently supports Python ``if``,
             ``elif``, ``else``, and ``for`` statements. Note that this feature requires an
             available TensorFlow installation. Please see the
-            `AutoGraph guide <https://docs.pennylane.ai/projects/catalyst/en/latest/dev/autograph.html>`__
-            for more information.
+            :doc:`AutoGraph guide <catalyst:dev/autograph>` for more information.
         target (str): the compilation target
         keep_intermediate (bool): Whether or not to store the intermediate files throughout the
             compilation. If ``True``, intermediate representations are available via the
             :attr:`~.QJIT.mlir`, :attr:`~.QJIT.jaxpr`, and :attr:`~.QJIT.qir`, representing
             different stages in the optimization process.
         verbosity (bool): If ``True``, the tools and flags used by Catalyst behind the scenes are
             printed out.
-        logfile (Optional[TextIOWrapper]): File object to write verbose messages to (default -
+        logfile (TextIOWrapper): File object to write verbose messages to (default is
             ``sys.stderr``).
-        pipelines (Optional(List[Tuple[str,List[str]]])): A list of pipelines to be executed. The
+        pipelines (List[Tuple[str,List[str]]]): A list of pipelines to be executed. The
             elements of this list are named sequences of MLIR passes to be executed. A ``None``
             value (the default) results in the execution of the default pipeline. This option is
             considered to be used by advanced users for low-level debugging purposes.
 
     Returns:
-        QJIT object.
+        QJIT: a class that, when executed, just-in-time compiles the decorated function
```

Comment:
Here, you want the LHS to be a type/class, and Sphinx will automatically try and cross reference it. We'll need to double check if this worked; if not, maybe `~.QJIT` or `catalyst:.QJIT` might work.

Reply:
Ah, got it working with `catalyst.QJIT` :) @maliasadi